crates/goose/src/prompt_template.rs

```diff
@@ -1,139 +1,238 @@
 use include_dir::{include_dir, Dir};
+use minijinja::{Environment, Error as MiniJinjaError, Value as MJValue};
+use once_cell::sync::Lazy;
 use serde::Serialize;
 use std::path::PathBuf;
-use tera::{Context, Error as TeraError, Tera};
+use std::sync::{Arc, RwLock};
+
+/// This directory will be embedded into the final binary.
+/// Typically used to store "core" or "system" prompts.
+static CORE_PROMPTS_DIR: Dir = include_dir!("$CARGO_MANIFEST_DIR/src/prompts");
+
+/// A global MiniJinja environment storing the "core" prompts.
+///
+/// - Loaded at startup from the `CORE_PROMPTS_DIR`.
+/// - Ideal for "system" templates that don't change often.
+/// - *Not* used for extension prompts (which are ephemeral).
+static GLOBAL_ENV: Lazy<Arc<RwLock<Environment<'static>>>> = Lazy::new(|| {
+    let mut env = Environment::new();
+
+    // Pre-load all core templates from the embedded dir.
+    for file in CORE_PROMPTS_DIR.files() {
+        let name = file.path().to_string_lossy().to_string();
+        let source = String::from_utf8_lossy(file.contents()).to_string();
+
+        // Since we're using 'static lifetime for the Environment, we need to ensure
+        // the strings we add as templates live for the entire program duration.
+        // We can achieve this by leaking the strings (acceptable for initialization).
+        let static_name: &'static str = Box::leak(name.into_boxed_str());
+        let static_source: &'static str = Box::leak(source.into_boxed_str());
+
+        if let Err(e) = env.add_template(static_name, static_source) {
+            eprintln!("Failed to add template {}: {}", static_name, e);
```

Comment:
nit: instead of `eprintln!` should this be `tracing::warn` or a different level?